metrics.go

```diff
@@ -35,11 +35,12 @@ func NewMetrics(namespace string) *Metrics {
 			Help:      "Counts the number of responses from cosmos separated by status code.",
 		}, statusCode)
 
-		retryAfterMS := promauto.NewGauge(prometheus.GaugeOpts{
+		retryAfterMS := promauto.NewHistogram(prometheus.HistogramOpts{
 			Namespace: namespace,
 			Subsystem: "cosmos",
 			Name:      "retry_after_ms",
 			Help:      "The time in milliseconds suggested by cosmos to wait before issuing the next query.",
+			Buckets: []float64{0,50,100,250,500,1000,5000,10000},
```

Comment:
As we are wondering if 1s or 2s or 3s is a good value it would be better to add more steps in this area.
```suggestion
			Buckets: []float64{0,50,100,250,500,1000,2000,3000,5000,7000,10000},
```